Dialogue changes about Deep Archaeology due to player's knowledge (#5940)
endless-sky/endless-sky: data/sheragi/archaeology missions.txt

		conversation
			`You breathe a sigh of relief as you land safely in the spaceport on <planet>. "Okay, what was that all about?" you ask.`
			`	"I honestly don't know," says Albert. "I can tell you what I suspect, though. The history books don't actually say when the Deep was first settled. The first documented expedition to the Deep was in the 24th century, and they found several human colonies already established in the region, claiming to date back to the very dawn of space flight. But I think the first settlers were brought there long before that. Brought there on alien ships."`
			branch "pug"
				or
					has "FWC Pug 2: offered"
					has "FW Pug 3: offered"
			choice
				`	"For what purpose?"`
					goto purpose
				`	"What aliens? The Quarg?"`
					goto aliens
			label pug
			choice
				`	"For what purpose?"`
					goto purpose

> 				`	"Wait, are you talking about the Pug?"`

Reviewer: ```suggestion
				`	"Do you think it was the Pug?"`
```
To me, the `Wait, are you talking about` feels like it implies that `<first>` thinks Albert's holding back info from them, which I think isn't what `<first>` would think in this situation. It really doesn't matter a whole lot though - your choice.

					goto surprise
			label purpose
			`	"I'm not sure," he says. "But I first became interested in the Deep while researching the Alpha Wars. I'm sure you know the history, how humanity was nearly enslaved by psychopathic 'super' humans that we ourselves had created by dabbling in genetic engineering. At that time, the Deep made a sudden, improbable technological leap that allowed them to construct weapons and ships far more powerful than what the Alphas had, and the war against them was won."`
				goto end
			label aliens
			`	"There are popular folk tales in the Deep about 'elves,'" says Albert. "Thin, pale, graceful creatures, with blue blood."`
			`	John frowns. "Hemocyanic blood. Not the Quarg, then. Their lungs are odd, but their blood uses hemoglobin for oxygen transport, just like ours."`
				goto end
			label surprise

> 			`	Albert and John look at each other confusedly before Albert says, "Who are the Pug, exactly?"`

Reviewer: The Pug invading human space is a fairly public event, right? Why wouldn't Albert and John already know about it? I can think of two ways you could change the branch in light of Albert and John probably already knowing about the Pug:
1. On one hand, you could just have them not know much about the Pug's description, and that wouldn't be much rewriting. However, I do find it hard to believe that they wouldn't have already known that too if they knew about the Pug.
2. In that case, maybe the best thing to do would be to cut the part where `<first>` explains who the Pug are and rewrite the later section to be Albert just explaining the connection between the Pug and folk tale aliens' descriptions.

I think I'd prefer the latter of these two options, but it's up to you.

Author: Of these options, I've elected to go with the latter. It's in the most recent commit.

			`	You go on to explain to them what you know about the Pug, their invasion into human space, and what you have heard of them interfering with human affairs in the Deep. You also give a brief description of what they look like, which puts a surprised look on both of their faces.`
			`	"Thin, pale, and with blue blood? That's precisely the description that we've had of these aliens based on folk tales," replies Albert.`

> 			`	"Well then we have good reason to believe they fit into our current knowledge of the timeline here. However, based on your description Captain <last>, they sound far too dangerous to attempt to seek out, especially after what we just fled from."`

Reviewer: ```suggestion
			`	"Well then, we have good reason to believe they fit into our current knowledge of the timeline here. However, based on your description Captain <last>, they sound far too dangerous to attempt to seek out, especially after what we just fled from."`
```
Instincts tell me there should be a comma there. I can look for a full explanation why if you want.

		conversation
			`You breathe a sigh of relief as you land safely in the spaceport on <planet>. "Okay, what was that all about?" you ask.`
			`	"I honestly don't know," says Albert. "I can tell you what I suspect, though. The history books don't actually say when the Deep was first settled. The first documented expedition to the Deep was in the 24th century, and they found several human colonies already established in the region, claiming to date back to the very dawn of space flight. But I think the first settlers were brought there long before that. Brought there on alien ships."`

> 			branch "pug"

Reviewer 2: ```suggestion
			branch pug
```

				or
					has "FWC Pug 2: offered"
					has "FW Pug 3: offered"
			choice
				`	"For what purpose?"`
					goto purpose
				`	"What aliens? The Quarg?"`
					goto aliens
			label pug
			choice
				`	"For what purpose?"`
					goto purpose
				`	"Do you think it was the Pug?"`
					goto surprise
			label purpose
			`	"I'm not sure," he says. "But I first became interested in the Deep while researching the Alpha Wars. I'm sure you know the history, how humanity was nearly enslaved by psychopathic 'super' humans that we ourselves had created by dabbling in genetic engineering. At that time, the Deep made a sudden, improbable technological leap that allowed them to construct weapons and ships far more powerful than what the Alphas had, and the war against them was won."`
				goto end
			label aliens
			`	"There are popular folk tales in the Deep about 'elves,'" says Albert. "Thin, pale, graceful creatures, with blue blood."`
			`	John frowns. "Hemocyanic blood. Not the Quarg, then. Their lungs are odd, but their blood uses hemoglobin for oxygen transport, just like ours."`
				goto end
			label surprise

> 			`	Albert and John look at each other confusedly for a second before Albert says, "The aliens that invaded human space?"`

Reviewer 3: ```suggestion
			`	Albert and John give confused looks at each other for a second before Albert says, "The aliens that invaded human space?"`
```

> 			`	John appears to think hard on the subject before saying, "Of course! I don't know how we didn't think of it. From the descriptions that I've heard about them, they almost exactly match the folk tales in the Deep about 'elves'."`

Reviewer: ```suggestion
			`	John appears to think hard on the subject before saying, "Of course! I don't know how we didn't think of it. From the descriptions that I've heard about them, they almost exactly match the folk tales in the Deep about 'elves.'"`
```
This is a weird preferred American punctuation thing. Don't blame ya for not knowing about it

			`	"Thin, pale, and blue blooded, yes? Well, I think you may have found our aliens, Captain <last>," replies Albert.`
			choice

> 				`	"That's great, but I think they're far too dangerous to be useful to us now."`

Reviewer 2: I don't really get how this choice follows from the previous dialog. The implication with this choice is that Albert and John were planning to contact whatever aliens brought humans to the Deep, yet that wasn't the case. Not really sure what to replace it with, though.

Author: That's a good point, seeing as the current state has them excitedly acknowledge this information only to divert it quickly. I suppose they could simply connect the dots and acknowledge it as being interesting, but not relevant, before moving on, though that may make it seem like the player's decision to bring up the Pug was pointless. I'll have to mull over my options for this dialogue for a bit.

Author: The latest commit provides a (hopefully) better structured option and dialogue flow.

					goto dangerous

> 				`	"From what I understand they've been around for a while, and have mentioned past interference in the Deep."`

Reviewer: ```suggestion
				`	"From what I understand, they've been around for a while, and have mentioned past interference in the Deep."`
```

> 			`	"Well then, we have good reason to believe they fit into our current knowledge of the timeline here. However, I do not think it is the greatest idea for us to get involved in a volatile alien species like that, especially after the danger we've just fled from."`

Reviewer: ```suggestion
			`	"Well then, we have good reason to believe they fit into our current knowledge of the timeline here. However, I don't think it's a good idea for us to get involved in a volatile alien species like that, especially after the danger we've just fled from."`
```
Optional. I thought adding some contractions improves the flow of the sentence.

> 			`	"Well then, we have good reason to believe they fit into our current knowledge of the timeline here. However, I don't think it's a good idea for us to get involved in a volatile alien species like that, especially after the danger we've just fled from."`

Reviewer 2: The last sentence here still didn't really feel like it fit to me, so here's an alternative.
```suggestion
			`	"Well then, if the Pug have admitted to 'interfering' with the Deep in the past then we have good reason to believe they fit into our current knowledge of the timeline here," Albert replies. "Unfortunately we can't just ask the Pug exactly what they've done or when they did it, but simply knowing of their presence here aids in our research."`
```

				goto final
			label dangerous

> 			`	"I've heard similar," says John, "but nevertheless it is at least useful to know the timeline of events here."`

Reviewer: ```suggestion
			`	"I've heard similar," says John, "but nevertheless, it is at least useful to know the timeline of events here."`
```
